Port to ESM for 45

oscfdezdz opened the pull request:
There aren't that many changes as the diff reflects, mostly because indentation. The main changes are imports and class headers.

![image](https://github.com/eliapasquali/power-profile-switcher/assets/42654671/63709afb-ea08-4bd9-b634-cae7f4082c14)

Fix #23

GlitchWitch commented:
I've tested your `esm` branch on Ubuntu 23.10 and encountered the following error when trying to open the settings dialog:

```
Gio.IOErrorEnum: Error opening file /home/user/.local/share/gnome-shell/extensions/power-profile-switcher@eliapasquali.github.io/ui/general.ui: No such file or directory

Stack trace:
  registerWidgetType@resource:///org/gnome/gjs/modules/core/overrides/Gtk.js:108:49
  _init/GObject.Object._classInit@resource:///org/gnome/gjs/modules/core/overrides/GObject.js:515:33
  _init/Gtk.Widget._classInit@resource:///org/gnome/gjs/modules/core/overrides/Gtk.js:75:31
  registerClass@resource:///org/gnome/gjs/modules/core/overrides/GObject.js:74:23
  fillPreferencesWindow@file:///home/user/.local/share/gnome-shell/extensions/power-profile-switcher@eliapasquali.github.io/prefs.js:35:31
  _loadPrefs@resource:///org/gnome/Shell/Extensions/js/extensionPrefsDialog.js:36:18
  async*_init@resource:///org/gnome/Shell/Extensions/js/extensionPrefsDialog.js:21:14
  ExtensionPrefsDialog@resource:///org/gnome/Shell/Extensions/js/extensionPrefsDialog.js:12:4
  OpenExtensionPrefsAsync@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:139:33
  async*LaunchExtensionPrefsAsync@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:126:14
  _handleMethodCall@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:373:35
  _wrapJSObject/<@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:408:34
  _init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:266:34
 
  ```
  
  If I manually copy the missing `ui` folder I encounter this error instead
  
  ```
  Error: Type name GeneralPreferences is already registered

Stack trace:
  registerGObjectType@resource:///org/gnome/gjs/modules/core/overrides/GObject.js:542:50
  registerWidgetType@resource:///org/gnome/gjs/modules/core/overrides/Gtk.js:92:39
  _init/GObject.Object._classInit@resource:///org/gnome/gjs/modules/core/overrides/GObject.js:515:33
  _init/Gtk.Widget._classInit@resource:///org/gnome/gjs/modules/core/overrides/Gtk.js:75:31
  registerClass@resource:///org/gnome/gjs/modules/core/overrides/GObject.js:74:23
  fillPreferencesWindow@file:///home/user/.local/share/gnome-shell/extensions/power-profile-switcher@eliapasquali.github.io/prefs.js:35:31
  _loadPrefs@resource:///org/gnome/Shell/Extensions/js/extensionPrefsDialog.js:36:18
  async*_init@resource:///org/gnome/Shell/Extensions/js/extensionPrefsDialog.js:21:14
  ExtensionPrefsDialog@resource:///org/gnome/Shell/Extensions/js/extensionPrefsDialog.js:12:4
  OpenExtensionPrefsAsync@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:139:33
  async*LaunchExtensionPrefsAsync@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:126:14
  _handleMethodCall@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:373:35
  _wrapJSObject/<@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:408:34
  _init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:266:34
  ```
  
---
  
Apart from the inability to configure the settings, the extenstion does otherwise work and switches between balanced mode on battery and performance on AC power. 

It's also worth noting that [gnome-extensions](https://github.com/mjakeman/extension-manager) tries to auto update this as the version number is listed as 3 whereas the latest published version has a version number of 12.

oscfdezdz commented:
Hi, try running `gnome-extensions pack --extra-source=ui` instead, I've updated the README in this branch with that. I've also changed the type name of the preferences that probably matched some other extension.

GlitchWitch commented:
That seemed to work! :tada:

GlitchWitch commented:
Upon further testing, the settings dialog only opens the first time. Subsquent opens cause the same general error.

```

oscfdezdz commented:
I think it should be fixed now! Thanks for testing 😄

GlitchWitch reviewed (approved):
Tested on a fresh install of Ubuntu 23.10, working perfectly!

GlitchWitch commented:
> I think it should be fixed now! Thanks for testing 😄

Yep, all good now! Thanks so much for this PR. Hopefully @eliapasquali will review and merge in soon :tada:

whosthegoss reviewed (approved):
lgtm

eliapasquali commented:
Thanks everyone for the amazing work and all the testing! Sorry for the quiet period.
